mitodl/mitxonline
# Cleanup and simplify configuration/localdev

#### Pre-Flight checklist


Part of #519 

#### What's this PR do?
This updates some of the docker setup as well as nginx configuration. Some key things:

- Removes `docker-compose.override.yml` and `docker-compose.notebook.yml` in favor of merging these all down to `docker-compose.yml`, which using [profiles](https://docs.docker.com/compose/profiles/) to make the notebook service optional.
- Removes bespoke `config/nginx.conf` that is only used locally in favor of a single nginx config file that already exists at `config/nginx.conf.erb`
- Creates a `Dockerfile` that extends the `nginx` docker image in order to emulate the compilation of the config ERB template.

#### How should this be manually tested?
Run `docker-compose build` and `docker-compose up`, the app should be reachable and you should see the output of the `20-compile-nginx-conf-erb.sh` script in the nginx container output.
